source/slang/slang-lower-to-ir.cpp

```diff
@@ -7924,86 +7951,66 @@ struct DeclLoweringVisitor : DeclVisitor<DeclLoweringVisitor, LoweredValInfo>
             return lowerGlobalConstantDecl(decl);
         }
 
-        IRType* varType = lowerType(context, decl->getType());
+        NestedContext nested(this);
+        auto subBuilder = nested.getBuilder();
+        auto subContext = nested.getContext();
 
-        auto builder = getBuilder();
+        IRGeneric* outerGeneric = nullptr;
+
+        // If we are static, then we need to insert the declaration before the parent.
+        // This tries to match the behavior of previous `lowerFunctionStaticConstVarDecl` functionality
+        if (isFunctionStaticVarDecl(decl))
```

Comment:
I am not sure if this is what I think it is.
This function name sounds like checking a static variable inside of a function like following.
```
bool MyFunc()
{
    static int s_vars[] = { 0, 1, 2 };
    static int s_unsized[];
```
I don't see this case in the test file.
Can we have a test for this case too?

Reply:
`static int s_vars[] = { 0, 1, 2 };` is probably a case we should allow. Let me update this PR to cover that.